modules/core/functions.php

```diff
@@ -357,27 +357,31 @@ function max_source_setting_callback($val) {
  */
 if (!hm_exists('save_user_settings')) {
 function save_user_settings($handler, $form, $logout) {
-    $user = $handler->session->get('username', false);
-    $path = $handler->config->get('user_settings_dir', false);
+    try {
+        $user = $handler->session->get('username', false);
+        $path = $handler->config->get('user_settings_dir', false);
 
-    if ($handler->session->auth($user, $form['password'])) {
-        $pass = $form['password'];
-    }
-    else {
-        Hm_Msgs::add('Incorrect password, could not save settings to the server', 'warning');
-        $pass = false;
-    }
-    if ($user && $path && $pass) {
-        $handler->user_config->save($user, $pass);
-        $handler->session->set('changed_settings', array());
-        if ($logout) {
-            $handler->session->destroy($handler->request);
-            Hm_Msgs::add('Saved user data on logout', 'info');
-            Hm_Msgs::add('Session destroyed on logout', 'info');
+        if ($handler->session->auth($user, $form['password'])) {
+            $pass = $form['password'];
         }
         else {
-            Hm_Msgs::add('Settings saved');
+            Hm_Msgs::add('Incorrect password, could not save settings to the server', 'warning');
+            $pass = false;
+        }
+        if ($user && $path && $pass) {
+            $handler->user_config->save($user, $pass);
+            $handler->session->set('changed_settings', array());
+            if ($logout) {
+                $handler->session->destroy($handler->request);
+                Hm_Msgs::add('Saved user data on logout', 'info');
+                Hm_Msgs::add('Session destroyed on logout', 'info');
+            }
+            else {
+                Hm_Msgs::add('Settings saved', 'info');
+            }
         }
+    } catch (Exception $e) {
```

Comment:
Catching here is OK but there are other places in Cypht that call save method on user config - you should handle the exception there as well.